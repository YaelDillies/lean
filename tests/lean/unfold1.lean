open tactic

meta definition rewriteH (Hname : name) : tactic unit :=
do do h ← get_local Hname,
   rewrite_target h,
   try reflexivity

example (l : list nat) : list.concat l [] = l :=
by do
  get_local `l >>= λ H, induction H [`h, `t, `iH],
  --
  dunfold_target [`list.concat],
  trace_state,
  trace "------",
  reflexivity,
  dunfold_target [`list.concat],
  trace_state,
  rewriteH `iH
