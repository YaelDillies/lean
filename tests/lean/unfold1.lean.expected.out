2 goals
⊢ list.nil = list.nil

h : ℕ,
t : list ℕ,
iH : t.concat list.nil = t
⊢ (h :: t).concat list.nil = h :: t
------
h : ℕ,
t : list ℕ,
iH : t.concat list.nil = t
⊢ h :: t.concat list.nil = h :: t
